Implementation/FindMyBLEDevice/FindMyBLEDevice/Services/CheckBluetoothAndLocationService.cs

```diff
@@ -0,0 +1,90 @@
+        {
+            running = false;
+        }
+
+        public void Start()
+        {
+            if (running) return;
+
+            running = true;
+            Task.Run(RunAsync);
+        }
+
+        public async Task RunAsync()
+        {
+            Console.WriteLine("Started Checking Services");
+
+            while (running)
+            {
+                try //just to be safe...
+                {
+
+                    try
+                    {
+                        var locationEnabled = platFormLocationService.IsLocationServiceEnabled();
+                        var bluetoothEnabled = App.Bluetooth.IsEnabled();
+
+                        string message = "If Bluetooth and GPS are not activated the correct functionality of this app cannot be guaranteed.\n";
+                        message += "Please enable the following service";
+                        message += !locationEnabled && !bluetoothEnabled ? "s:\n" : ":\n";
+                        message += !bluetoothEnabled ? "\n - Bluetooth" : "";
+                        message += !locationEnabled ? "\n - GPS" : "";
+
+                        // check is only valid for iOS devices, for Android the permission service always returns true
+                        // This is done because requesting permissions for Android works completely different and it was not possible to implement it in the same way
+                        bool hasBluetoothPermission = permissionService.checkBluetoothPermission();
+                        bool hasLocationPermission = permissionService.checkLocationPermission();
+
+                        if (!hasBluetoothPermission || !hasLocationPermission)
+                        {
+                            string msg = "The app is not functioning correctly because the permissions are not granted. Please navigate to the settings app to grant the required permissions";
+                            await Xamarin.Forms.Device.InvokeOnMainThreadAsync(async () =>
+                            {
+                                await Application.Current.MainPage.DisplayAlert("Attention", msg, "Ok", "Cancel");
+                            });
+                        }
+                        else if (!locationEnabled || !bluetoothEnabled)
+                        {
+                            await Xamarin.Forms.Device.InvokeOnMainThreadAsync(async () =>
+                            {
+                                await Application.Current.MainPage.DisplayAlert("Attention", message, "Ok");
+                            });
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine(e.Message);
+                    }
+
+                    await Task.Delay(checkInterval);
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.ToString());
+                }
```

Comment:
I don't like the two directly nested try-catch-clauses. I would suggest to simply remove the inner one. 
You could move the await Task.Delay() to the top, but then the service would never get to actually check for permissions if for some reason the delay always fails / gets interrupted. I would rather have it the other way round, so the delay doesn't happen if something above it fails, the loop immediately restarts and tries again.